module/zfs/zfs_rlock.c

```diff
@@ -389,7 +393,7 @@ zfs_rangelock_enter_reader(rangelock_t *rl, locked_range_t *new)
 retry:
 	prev = avl_find(tree, new, &where);
 	if (prev == NULL)
-		prev = (locked_range_t *)avl_nearest(tree, where, AVL_BEFORE);
+		prev = (void *)avl_nearest(tree, where, AVL_BEFORE);
```

Comment:
To keep under the length and preserve the cast I'd suggest this:

```c
        prev = avl_find(tree, new, &where);
        if (prev == NULL) {
                prev = (zfs_locked_range_t *)
                    avl_nearest(tree, where, AVL_BEFORE);
        } 
```

Reply:
The cast provides no type checking value, but if you prefer the aesthetic.

Reply:
I don't think that any cast is needed (or desired) here (or below), since `avl_nearest()` returns `void*`.

Reply:
Removing them entirely is OK with me.  They provide some additional documentation but I don't think in this case it's helpful.